themes/gokarna/static/js/main.js

```diff
@@ -46,10 +34,27 @@ window.addEventListener('scroll', () => {
     }
 });
 
+function setThemeByUserPref() {
+    darkThemeCss = document.getElementById("dark-theme");
+    const savedTheme = localStorage.getItem(THEME_PREF_STORAGE_KEY) || 
+        (window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark': 'light');
+    const darkThemeToggles = document.querySelectorAll('.dark-theme-toggle');
+    setTheme(savedTheme, darkThemeToggles);
+    darkThemeToggles.forEach(el => el.addEventListener('click', (event) => {
+        toggleIcon = event.currentTarget.querySelector("a svg.feather");
+        if (toggleIcon.classList[1] === THEME_TO_ICON_CLASS.dark) {
+            setTheme('light', [event.currentTarget]);
+        } else if (toggleIcon.classList[1] === THEME_TO_ICON_CLASS.light) {
+            setTheme('dark', [event.currentTarget]);
+        }
+    }));
+}
 
-function setTheme(themeToSet) {
+function setTheme(themeToSet, targets) {
```

Comment:
Since you have refactored javascript, ismain themeMode bhi incorporate karle.
Default: `auto`, else `light` and `dark`, this argument will come from the config.toml

Reply:
So will we remove the option to toggle theme if it's set to `light` or `dark` explicitly? If yes, then it needs to be done in a separate PR. Doesn't make sense to fit it in here.